### PR TITLE
fix: sherlock

### PR DESCRIPTION
**Pull request by Gevarist**

### CODE REVIEW

**Comment by Gevarist** [comment]
this PR intend to resolve sherlock issue [#83](https://github.com/sherlock-audit/2023-06-arrakis-judging/issues/83), [#26](https://github.com/sherlock-audit/2023-06-arrakis-judging/issues/26) and [#25](https://github.com/sherlock-audit/2023-06-arrakis-judging/issues/25).

[#83](https://github.com/sherlock-audit/2023-06-arrakis-judging/issues/83) : we add an outdated variable for each price feed.
[#26](https://github.com/sherlock-audit/2023-06-arrakis-judging/issues/26) : we add a extra pool deviation check after rebalance.
[#25](https://github.com/sherlock-audit/2023-06-arrakis-judging/issues/25) : we add a cooldown period during when rebalance cannot happen.

File: contracts/SimpleManager.sol [hunk 1]
```diff
@@ -136,14 +145,22 @@ contract SimpleManager is OwnableUpgradeable {
...
+        require(
+            vaultInfo.lastRebalanceTimestamp + vaultInfo.coolDownPeriod <
+                block.timestamp, // solhint-disable-line not-rely-on-time
+            "CP"
+        );
+
+        {
+            VaultInfo storage vaultInfoS = vaults[vault_];
+            // solhint-disable-next-line not-rely-on-time
+            vaultInfoS.lastRebalanceTimestamp = block.timestamp;
```
Comment (jack-the-pug) [inline contracts/SimpleManager.sol:157]: Why not change to `vaults[vault_].lastRebalanceTimestamp = block.timestamp;` for simplicity?

File: contracts/SimpleManager.sol [hunk 2]
```diff
@@ -315,6 +309,70 @@ contract SimpleManager is OwnableUpgradeable {
...
+
+            (sqrtPriceX96, , , , , , ) = pool.slot0();
+
+            uint256 poolPrice;
+
+            if (sqrtPriceX96 <= type(uint128).max) {
+                poolPrice = FullMath.mulDiv(
+                    sqrtPriceX96 * sqrtPriceX96,
+                    10 ** token0Decimals_,
+                    2 ** 192
```
Comment (jack-the-pug) [inline contracts/SimpleManager.sol:354]: Consider changing to `1 << 192` for consistency with `1 << 128`.

File: contracts/SimpleManager.sol [hunk 3]
```diff
@@ -154,48 +171,15 @@ contract SimpleManager is OwnableUpgradeable {
...
-                poolPrice,
-                oraclePrice,
-                vaultInfo.maxDeviation,
-                token1Decimals
-            );
-
-            checked[increment] = rebalanceParams_.mints[i].range.feeTier;
-            increment++;
-        }
+        _checkPoolsPrices(
```
Comment (jack-the-pug) [inline contracts/SimpleManager.sol:174]: Do we still need to conduct a price check prior to the rebalance, considering there is already a post-rebalance check? Seems necessary, the post-rebalance check should be enough.

File: contracts/SimpleManager.sol (hunk 1 above)
Comment (Gevarist) [reply contracts/SimpleManager.sol:157]: Sure, will change that.

File: contracts/SimpleManager.sol (hunk 2 above)
Comment (Gevarist) [reply contracts/SimpleManager.sol:354]: Yes will make the change.

**Review by kassandraoftroy — APPROVED** [review]
LGTM

**Pull request merged**